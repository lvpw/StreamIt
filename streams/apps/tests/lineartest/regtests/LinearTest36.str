/** 
 * Very simple program for LinearRedundancyReplacement
 * testing.
 **/
void->void pipeline LinearTest36 {
  add FloatSource();
  add SimpleFIRFilter();
  add FloatSink();
}

void->float stateful filter FloatSource {
  int x;
  init {
    x = 0;
  }
  work push 1 {
    push(x);
    x = (x + 1)%4;
  }
}

float->void filter FloatSink {
  work pop 1 {
    println(pop());
  }
}

/** 
 * This filter has the linear form [2 3 2]'.
 **/
float->float filter SimpleFIRFilter() {
  work peek 3 pop 1 push 1 {
    push(2*peek(0) + 3*peek(1) + 2*peek(2));
    pop();
  }
}
