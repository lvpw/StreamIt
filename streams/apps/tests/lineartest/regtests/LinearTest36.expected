SimpleFIRFilter.matrix
[[2.0+0.0i]
 [3.0+0.0i]
 [2.0+0.0i]]
SimpleFIRFilter.vector
[[0.0+0.0i]]
